asapdiscovery-docking/asapdiscovery/docking/scripts/single_target_docking.py

```diff
@@ -0,0 +1,416 @@
+)
+
+parser.add_argument(
+    "--omega",
+    action="store_true",
+    help="Use Omega conformer enumeration.",
+)
+
+parser.add_argument(
+    "--hybrid",
+    action="store_true",
+    help="Whether to only use hybrid docking protocol in POSIT.",
+)
+
+parser.add_argument(
+    "--num_poses",
+    type=int,
+    default=1,
+    help="Number of poses to return from docking.",
+)
+
+parser.add_argument(
+    "--gat",
+    action="store_true",
+    help="Whether to use GAT model to score docked poses.",
+)
+
+
+def main():
+    args = parser.parse_args()
+
+    # setup output directory
+    output_dir = Path(args.output_dir)
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    logname = "prep-mcs-dock-single-target"
+    # setup logging
+    logger = FileLogger(logname, path=output_dir, stdout=True).getLogger()
+    logger.info(f"Start single target prep+docking at {datetime.now().isoformat()}")
+    logger.info(f"Output directory: {output_dir}")
+
+    # openeye logging handling
+    errfs = oechem.oeofstream(str(output_dir / f"openeye-{logname}-log.txt"))
+    oechem.OEThrow.SetOutputStream(errfs)
+    oechem.OEThrow.SetLevel(oechem.OEErrorLevel_Debug)
+
+    if args.debug:
+        logger.info("Running in debug mode. enabling --verbose and disabling --cleanup")
+        logger.info(f"Input arguments: {args}")
+        args.verbose = True
+        args.cleanup = False
+
+    # paths to remove if not keeping intermediate files
+    intermediate_files = []
+
+    # parse input molecules
+    if is_valid_smiles(args.mols):
+        logger.info(
+            f"Input molecules is a single SMILES string: {args.mols}, using title {args.title}"
+        )
+        # hash the smiles to generate a unique title and avoid accidentally caching different outputs as the same
+        if not args.title:
+            logger.info(
+                "No title provided, MD5 hashing SMILES string to generate title, consider providing a title with --title"
+            )
+            args.title = (
+                "TARGET_MOL-" + hashlib.md5(args.mols.encode("utf-8")).hexdigest()
+            )
+
+        exp_data = [ExperimentalCompoundData(compound_id=args.title, smiles=args.mols)]
+    else:
+        mol_file_path = Path(args.mols)
+        if mol_file_path.suffix == ".smi":
+            logger.info(f"Input molecules is a SMILES file: {args.mols}")
+            exp_data = oe_load_exp_from_file(args.mols, "smi")
+        elif mol_file_path.suffix == ".sdf":
+            logger.info(f"Input molecules is a SDF file: {args.mols}")
+            exp_data = oe_load_exp_from_file(args.mols, "sdf")
+        else:
+            raise ValueError(
+                f"Input molecules must be a SMILES file, SDF file, or SMILES string. Got {args.mols}"
+            )
+
+    logger.info(f"Loaded {len(exp_data)} molecules.")
+
+    if args.verbose:
+        for exp in exp_data:
+            logger.info(f"Loaded molecule {exp.compound_id}: {exp.smiles}")
```

Comment:
why not use `logger.debug` and then set the logger level to `debug` only if the verbose flag is passed? that would be cool